### ISSUE TITLE
dart2native default output filename should be `file.exe`, not `file.dart.exe`

### ISSUE THREAD

**Issue by timsneath** [dart-lang/sdk#38530]
See command line output below:
```
C:\src\dart_console>\dart\bin\dart2native example\kilo.dart
Generating AOT kernel dill.
Generating AOT snapshot.
Generating executable.
Generated: c:\src\dart_console\example\kilo.dart.exe
```

We should strip the `.dart` extension from the input and the output should be `kilo.exe`. This is in keeping with other compilers and (I would aver) developers' expectations.

**Comment by mit-mit** [comment]
Yes, that would be better!

**Comment by cskau-g** [comment]
I can add a small convenience feature to detect and remove a '.dart' prefix from the default output file name.
Just bear in mind that you can already set the output filename via `--output=my.exe` if you don't want it to pick something for you.
The problem with trying to be too smart about the auto-generated output filename is that users are free to give any filename as input, so simply replacing '.dart' with '.exe' won't always give you what you want.

E.g. what do we pick when someone runs:
- `dart2native my.dart.txt`
- `dart2native my_dart`
- `dart2native .dart`
etc. ?

**Comment by mit-mit** [comment]
I don't think we have to be super clever. It's a very resonable assumption that 99.x% of all inputs will be `<base file name>.dart`, as that's what you need to follow to get editor support.

So I would simply see if the file ends with `.dart` and if so remove that from the output filename.

**Comment by timsneath** [comment]
I think it would actually be reasonable to assert if `.dart` is *not* the
extension of the input. Other compilers certainly expect that the extension
matches the input:
$ g++ test2.cpp
$ cp test2.cpp test2.txt
$ g++ test2.txt
ld: warning: ignoring file test2.txt, building for macOS-x86_64 but
attempting to link with file built for unknown-unsupported file format (
0x2F 0x2F 0x20 0x68 0x65 0x6C 0x6C 0x6F 0x2E 0x63 0x70 0x70 0x0A 0x23 0x69
0x6E )
Undefined symbols for architecture x86_64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see
invocation)

**Comment by cskau-g** [comment]
Fixed in bb600e9.
